app/services/transfer_product_from_stock_service.rb

```diff
@@ -0,0 +1,17 @@
+class TransferProductFromStockService
+  def initialize(product:, location:, second_location:, quantity:)
+    @product = product
+    @location = location
+    @second_location = second_location
+    @quantity = quantity.to_i
+  end
+
+  def call
+    @stock_1 = Stock.find_by(location: @location, product: @product)
+    @stock_1.quantity -=  @quantity
+    @stock_1.save
+    @stock_2 = Stock.find_by(location: @second_location, product: @product)
+    @stock_2.quantity += @quantity
+    @stock_2.save
```

Comment:
I think this should be run inside a transaction. Imagine the `@second_location` contains incorrect id. If this happens then the code will decrease the stock in the first location and then fail, leaving the stocks inconsistent.